runtime/pkg/rilltime/rilltime_test.go

```diff
@@ -0,0 +1,82 @@
+package rilltime
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Resolve(t *testing.T) {
+	now := parseTestTime(t, "2024-08-09T10:32:36Z")
+	maxTime := parseTestTime(t, "2024-08-06T06:32:36Z")
+	watermark := parseTestTime(t, "2024-08-05T06:32:36Z")
+	testCases := []struct {
+		timeRange string
+		start     string
+		end       string
+	}{
+		// Earliest = 2023-08-09T10:32:36Z, Latest = 2024-08-06T06:32:36Z, = Now = 2024-08-09T10:32:36Z
+		{`m : |s|`, "2024-08-09T10:32:00Z", "2024-08-09T10:32:37Z"},
+		{`m : s`, "2024-08-09T10:32:00Z", "2024-08-09T10:32:37Z"},
```

Comment:
Shouldn't it be:
```
		{`m : |s|`, "2024-08-09T10:32:00Z", "2024-08-09T10:32:36Z"},
```
like it was before? I.e. `|s|` leads to `:36Z` and `s` leads to `:37Z`.

The idea being that `|s|` indicates that only completed bins should be included – but if `maxTime=...:36Z`, more events may yet appear with `time=...:36Z`. And callers who are good with bins that have potentially partial data should use `m : s`.

Reply:
Hmmm. So the idea is that our bins are `ts >= 0 and ts < next 0`? So a data point at the boundary is actually part of next grain. That is why `|s|` will not include it since it is incomplete. We should make sure we explain all these things in our docs, will note it down.

Reply:
Yeah that's the idea!